app/components/connectionForm.js

```diff
@@ -13,7 +13,11 @@ export default class Connect extends Component {
     return (
       <div>
         <input ref="ip" id="ip" name="ip" placeholder="Ip Eg: 12.0.0.1/localhost"/>
-        <button type="button" className="btn btn-success" onClick={this.handleConnect}>Connect</button>
+        <button type="button" className="btn btn-success" onClick={this.handleConnect}>
+          {
+            this.props.status === 'CONNECTING' ? 'Connecting...' : 'Connect'
```

Comment:
We already have a state in redux store and it won't make sense to duplicate
here as component Level state.
A prop change or a state change will be same here.

On Wed, Mar 23, 2016 at 4:52 PM, Yuva <notifications@github.com> wrote:

> In app/components/connectionForm.js
> <https://github.com/code-mancers/rbkit-electron-redux/pull/9#discussion_r57143470>
> :
>
> > @@ -13,7 +13,11 @@ export default class Connect extends Component {
> >      return (
> >        <div>
> >          <input ref="ip" id="ip" name="ip" placeholder="Ip Eg: 12.0.0.1/localhost"/>
> > -        <button type="button" className="btn btn-success" onClick={this.handleConnect}>Connect</button>
> > +        <button type="button" className="btn btn-success" onClick={this.handleConnect}>
> > +          {
> > +            this.props.status === 'CONNECTING' ? 'Connecting...' : 'Connect'
>
> yeah, not sure if it makes sense since you are using redux.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/code-mancers/rbkit-electron-redux/pull/9/files/b5077fdb7adf104551977f5c9565030567154b5e#r57143470>
>